### PR TITLE
Added information about repo in readme

### PR DESCRIPTION
**Pull request by JRosenkranz**
Updated readme with relevant information

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,34 @@
...
+
+* `fms_extras/models/` - Pure pytorch implementations of popular model architectures, without requiring any specific common interface beyond `nn.Module`. Each model configuration is registered with `fms.models.register_model()` so that instances can be obtained through `fms.models.get_model('architecture', 'variant', '/path/to/data')`. Each model can also register sources/formats/versions of data to load (e.g. checkpoints provided by meta, HF, or trained from this repo). Users of the repo (e.g. `fms-extras`) can register their own model architectures as well.
+* `fms_extras/models/hf/` - Adapters that compose our native PyTorch FMS model architecture implementations in HF-compatible wrapper interfaces. Each FMS model implements an adapter, and adapted instances are obtained via `fms.models.hf.to_hf_api(model)`
+* `fms_extras/utils/` - Other operators useful in working with LLMs. These include a `generate()` function, `Tensor` subclasses, code for dealing with LLM checkpoints that might be saved/sharded in a variety of formats, tokenization code, and various other useful helper functions.
```
Comment (nairbv) [inline README.md]: looks like you copy/pasted this from fms and it's retaining some items not relevant here. e.g. 'extras' doesn't have it's own `generate` function (though maybe a `speculative_generate()`?)

Comment (JRosenkranz) [reply README.md]: Yes, we have kept the structure the same here, so wanted to keep the same, but will fix with the relevant information

Comment (JRosenkranz) [reply README.md]: updated in latest commit

File: README.md [hunk 2]
```diff
@@ -1 +1,34 @@
...
+
+* `fms_extras/models/` - Pure pytorch implementations of popular model architectures, without requiring any specific common interface beyond `nn.Module`. Each model configuration is registered with `fms.models.register_model()` so that instances can be obtained through `fms.models.get_model('architecture', 'variant', '/path/to/data')`. Each model can also register sources/formats/versions of data to load (e.g. checkpoints provided by meta, HF, or trained from this repo). Users of the repo (e.g. `fms-extras`) can register their own model architectures as well.
+* `fms_extras/models/hf/` - Adapters that compose our native PyTorch FMS model architecture implementations in HF-compatible wrapper interfaces. Each FMS model implements an adapter, and adapted instances are obtained via `fms.models.hf.to_hf_api(model)`
+* `fms_extras/utils/` - Other operators useful in working with LLMs. These include a `speculative_generate()` function, `PagedKVCacheManager` class for easy-to-use kv-cache management with paged attention kernels, etc.
+* `scripts/` - Various scripts for inference, benchmarking, and evaluation, as well as an entry-point for tuning/training.
```
Comment (nairbv) [inline README.md]: I think the copy here is still inaccurate for this repo. E.g. we don't have an evaluation script or an entry point for training here. There are probably other differences as well if we check carefully.

we should try to call out what's actually different/unique about this repo so we're not left updating this copy of the comment anytime fms's readme changes.

Comment (JRosenkranz) [reply README.md]: Fixed in latest commit

File: README.md [hunk 3]
```diff
@@ -1 +1,35 @@
...
+3. `PagedLLaMA`: a LLaMA implementation that uses paged-attention in Multi-Head Attention. This model is compilable without graph breaks.
+4. `speculative generation`: a reference implementation of speculative generate using PagedKVCacheManager and MLPSpeculator
+
+## Structure and contents of this Repository
+
+This repo follows a similar structure to that of [foundation-model-stack](https://github.com/foundation-model-stack/foundation-model-stack)
+
+* `fms_extras/models/` - Pure pytorch implementations of popular model architectures, without requiring any specific common interface beyond `nn.Module`. Each model configuration is registered with `fms.models.register_model()` so that instances can be obtained through `fms.models.get_model('architecture', 'variant', '/path/to/data')`. Each model can also register sources/formats/versions of data to load (e.g. checkpoints provided by meta, HF, or trained from this repo). Users of the repo (e.g. `fms-extras`) can register their own model architectures as well.
```
Comment (nairbv) [inline README.md]: "Users of the repo (e.g. `fms-extras`) can register their own model architectures as well."

**Review by nairbv — APPROVED** [review]

**Pull request merged**